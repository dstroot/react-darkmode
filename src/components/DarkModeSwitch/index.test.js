import React from "react";
import { render, fireEvent } from "@testing-library/react";

// component to test
import { DarkModeSwitch } from ".";

// tests
describe("DarkModeSwitch", () => {
  it("renders a dark mode switch", () => {
    const { asFragment } = render(<DarkModeSwitch />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("should toggle dark mode", () => {
    const { container } = render(<DarkModeSwitch />);
    const darkModeSwitch = container.querySelector('[id="react-switch-new"]');

    // switch to dark mode
    fireEvent.click(darkModeSwitch);

    let val = localStorage.getItem("theme");
    let theme = JSON.parse(val);
    expect(theme).toEqual("dark");

    // switch to light mode
    fireEvent.click(darkModeSwitch);

    val = localStorage.getItem("theme");
    theme = JSON.parse(val);
    expect(theme).toEqual("light");
  });

  it("should read the browser preference", () => {
    // noop - can't test for window in test env
  });
});
