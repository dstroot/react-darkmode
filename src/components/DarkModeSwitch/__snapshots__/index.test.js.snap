// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DarkModeSwitch renders a dark mode switch 1`] = `
<DocumentFragment>
  <input
    class="react-switch-checkbox"
    id="react-switch-new"
    type="checkbox"
  />
  <label
    class="react-switch-label"
    for="react-switch-new"
  >
    <span
      class="react-switch-button"
    />
  </label>
</DocumentFragment>
`;
